include/glow/Optimizer/CompilationContext.h

```diff
@@ -19,20 +19,90 @@
 #include "glow/Backends/BackendOptions.h"
 #include "glow/Graph/PlaceholderBindings.h"
 #include "glow/Quantization/Base/Base.h"
+#include "glow/Support/Error.h"
 
 namespace glow {
 
+/// Configuration for different precision modes.
+struct PrecisionConfiguration {
+  /// Enum for what kind of transformation should be done for Quantization.
+  enum class QuantizationMode {
+    None,     /// Perform no transformations for quantization.
+    Quantize, /// Quantize the graph using previously gathered statistics.
+    Profile,  /// Add profiling nodes for quantization statistics gathering.
+  } quantMode{QuantizationMode::None};
+
+  /// Configuration for Quantization.
+  quantization::QuantizationConfiguration quantConfig;
+
+  /// Whether to convert the FloatTy to Float16Ty in the Function.
+  bool convertToFP16{false};
+
+  /// Used during Quantization, Profiling, and convertToFP16.
```

Comment:
Since this comment could be more helpful, probably mentioning something about this being kinds to exclude or something like that